### PR TITLE
HDS-2027: improve header docs

### PR DESCRIPTION
**Pull request by NikoHelle**
## Description

Added missing docs for header components.

## Related Issue

Closes [HDS-2027](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2027)

## How Has This Been Tested?

[Demo](https://city-of-helsinki.github.io/hds-demo/hds-2027-header-docs/components/header/code/)

## Add to changelog
- No need to document small documentation update

[HDS-2027]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/components/header/code.mdx [hunk 1]
```diff
@@ -709,27 +708,30 @@ import { Header, Logo, logoFi } from 'hds-react';
...
+| `logoAriaLabel`                       | Aria-label for the logo.                                                                     | `string`                                      | -             |
+| `logoHref`                            | URL for the logo.                                                                            | `string`                                      | -             |
+| `menuButtonAriaLabel`                 | Aria-label for the menu button.                                                              | `string`                                      | -             |
+| `onLogoClick`                         | Callback function for when the logo is clicked.                                              | `MouseEventHandler`                           | -             |
+| `onMenuButtonClick`                   | Callback function for when the menu button is clicked.                                       | `MouseEventHandler`                           | -             |
```
[the remark this answers is not in the document]
Comment (harriplappalainen) [reply site/src/docs/components/header/code.mdx:727]: We discussed this with the team and these props names in the docs will be fixed in [this ticket's PR](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2017).

File: site/src/docs/components/header/code.mdx [hunk 2]
```diff
@@ -742,24 +744,30 @@ import { Header, Logo, logoFi } from 'hds-react';
...
+| `children`                                   | Additional languages to display in the dropdown.                                                                                                                                           | `JSX.Element`                                                                                   | -             |
+| `languages`                                  | Array of languages to display in the Header.ActionBar.                                                                                                                                     | `[LanguageOption]`                                                                              | -             |
+| `sortLanguageOptions`                        | Function for sorting language options into primary and secondary. By default, all primary `LanguageOptions` as shown. If a secondary `LanguageOption` is selected, only that one is shown. | `(options: LanguageOption[], selectedLanguage: string) => [LanguageOption[], LanguageOption[]]` | -             |
```
[the remark this answers is not in the document]
Comment (harriplappalainen) [reply site/src/docs/components/header/code.mdx:751]: We discussed this with the team and these props names in the docs will be fixed in [this ticket's PR](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2017). So no need to do anything here.

**Review by harriplappalainen — COMMENTED** [review]
Good work 👍 Some comments below.

File: site/src/docs/components/header/code.mdx [hunk 3]
```diff
@@ -773,8 +781,12 @@ import { Header, Logo, logoFi } from 'hds-react';
 | `dropdownLinkClassName`           | Additional classNames to apply to the dropdown links.                       | `string`          | -                                                       |
 | `dropdownLinks`                   | Array of NavigationLink components to render in a dropdown.                 | `[<Header.Link>]` | -                                                       |
 | `href`                            | Hypertext Reference of the link.                                            | `string`          | -                                                       |
+| `id`                              | Id for the link.                                                            | `string`          | -                                                       |
+| `index`                           | Element index given by parent mapping.                                      | `number`          | -                                                       |
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply site/src/docs/components/header/code.mdx]: Oh yes, did not notice the `@internal`. Removed.

**Comment by NikoHelle** [comment]
Demo updated.

**Review by harriplappalainen — APPROVED** [review]
This seems to do what was wanted 👍 Maybe @laurakarhu should review this as well.

File: site/src/docs/components/header/index.mdx [hunk 4]
```diff
@@ -248,6 +248,16 @@ Header.LanguageSelector offers the languages for the whole service. It is recomm
 - See example for Header.LanguageSelector <Link openInNewTab openInNewTabAriaLabel="Opens in a new tab" href="/storybook/react/?path=/story/components-header--with-full-features">in Storybook</Link>.
 - Under large sized screens (992px and down) the Header.LanguageSelector moves under the Header.ActionBar.
```
Comment (laurakarhu) [inline site/src/docs/components/header/index.mdx:250]: Should we describe also the default language selector with somehting like:

Use the Header.LanguageSelector when:
- Responsive mobile view is needed
- Some languages are shown outside the language dropdown with isPrimary: true, dropdown for secondary languages is left out if there are not any other languages
- There is a need for additional languages

**Review by laurakarhu — CHANGES_REQUESTED** [review]
Good additions. I would maybe make it even more clear how the normal language selector is used, as that was the original issue, the projects assumed simple version was needed for using only 3 default languages.

File: site/src/docs/components/header/index.mdx (hunk 4 above)
Comment (NikoHelle) [reply site/src/docs/components/header/index.mdx:252]: https://city-of-helsinki.github.io/hds-demo/hds-2027-header-docs/components/header/

**Review by laurakarhu — APPROVED** [review]
Thank you, looking good now!

**Pull request merged**